File: src/main/java/io/javalin/lifecycle/Event.java
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 *
 */

package io.javalin.lifecycle;

import io.javalin.Javalin;

public class Event {

    public enum Type {
        SERVER_STARTING,
        SERVER_STARTED,
        SERVER_STOPPING,
        SERVER_STOPPED
    }

    public Type eventType;
    public Javalin javalin;

    public Event(Type eventType) {
        this.eventType = eventType;
        this.javalin = null;
    }

    public Event(Type eventType, Javalin javalin) {
        this.eventType = eventType;
        this.javalin = javalin;
    }

}

File: src/main/java/io/javalin/lifecycle/Event.kt
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin.lifecycle

import io.javalin.Javalin

data class Event(var eventType: Type, var javalin: Javalin? = null) {
    enum class Type {
        SERVER_STARTING,
        SERVER_STARTED,
        SERVER_START_FAILED,
        SERVER_STOPPING,
        SERVER_STOPPED
    }
}
